### ISSUE TITLE
Feature Request: http functions could handle HTTP REDIRECT

### ISSUE THREAD

**Issue by joysfera** [nodemcu/nodemcu-firmware#1366]
The http.get() and other useful functions from http module could handle HTTP 30x return code (redirect) automatically (= repeat the request to the new URL provided by server in the Location header). 

Alternatively the http functions should return also the received http headers (apart from status and body) so that users could issue another request to redirected URL by themselves (when they find out that status code was 30x).

Reason for this is that we can upload IoT data directly to Google Sheets (for example) but there's a http redirect that needs to be processed.

Frankly, I'd love to get the http response headers in the http.get() callback for now. Parsing out the redirected URL from Location header would be piece of cake.

**Comment by joysfera** [comment]
What a bad luck: I have spent this nice shiny summer Sunday writing a replacement for http module in Lua that handles redirects internally. I got it working, it can connect to "script.google.com" and send a value directly to Google Sheets. Great. Well, almost. When I plug this function into my rather complex program with just 16 kB on node.heap() the net.socket:connect() does NOT connect. There's no error message, no crash, nothing. The rest of application continues to work normally, it just doesn't connect. Is this small memory issue? I think so and I'm real unhappy with this.
